aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -111,6 +130,91 @@ fn native_get_script_hash(
     )])
 }
 
+fn native_sender_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SENDER_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        Ok(smallvec![Value::address(transaction_context.sender())])
+    } else {
+        Err(SafeNativeError::Abort {
+            abort_code: error::invalid_state(abort_codes::E_TRANSACTION_CONTEXT_NOT_AVAILABLE),
+        })
+    }
+}
+
+fn native_secondary_signers_internal(
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(TRANSACTION_CONTEXT_SECONDARY_SIGNERS_BASE)?;
+
+    let transaction_context_opt = get_transaction_context_opt_from_context(context);
+    if let Some(transaction_context) = transaction_context_opt {
+        let secondary_signers = transaction_context.secondary_signers();
```

Comment:
Does this include the gas fee payer if it's separate?

Reply:
This does not include the gas payer. `fee_payer` is a separate field in a signed transaction: https://github.com/aptos-labs/aptos-core/blob/ce593c4b0c9f066c821b80c403d6b5f8ec12263b/types/src/transaction/mod.rs#L310-L313